website/addons/github/static/githubFangornConfig.js

```diff
@@ -8,67 +8,118 @@ var URI = require('URIjs');
+        var deleteList = [];
+        var noDeleteList = [];
+        var mithrilContentMultiple;
+        var mithrilButtonsMultiple;
+        items.forEach(function(item, index, arr){
+            if(!item.data.permissions.edit){
+                canDelete = false;
+                noDeleteList.push(item);
+            } else {
+                deleteList.push(item);
+            }
+        });
+        // If all items can be deleted
+        if(canDelete){
+            mithrilContentMultiple = m('div', [
+                    m('h3.break-word', 'Delete multiple files?'),
                     m('p', 'This action is irreversible.'),
-                    parent.children.length < 2 ? m('p', 'If a folder in Github has no children it will automatically be removed.') : ''
+                    deleteList.map(function(item){
+                        return m('.fangorn-canDelete.text-success', item.data.name);
+                    })
                 ]);
-            var mithrilButtons = m('div', [
-                    m('button', { 'class' : 'btn btn-default m-r-md', onclick : function() { cancelDelete.call(tb); } }, 'Cancel'),
-                    m('button', { 'class' : 'btn btn-success', onclick : function() { runDelete.call(tb); }  }, 'OK')
+            mithrilButtonsMultiple =  m('div', [
+                    m('span.tb-modal-btn', { 'class' : 'text-primary', onclick : function() { cancelDelete(); } }, 'Cancel'),
+                    m('span.tb-modal-btn', { 'class' : 'text-danger', onclick : function() { runDeleteMultiple.call(tb, deleteList); }  }, 'Delete All')
                 ]);
-            tb.modal.update(mithrilContent, mithrilButtons);
         } else {
-            item.notify.update('You don\'t have permission to delete this file.', 'info', undefined, 3000);
+            mithrilContentMultiple = m('div', [
+                    m('h3.break-word', 'Delete multiple files?'),
+                    m('p', 'Some of these files can\'t be deleted but you can delete the ones highlighted with green. This action is irreversible.'),
+                    deleteList.map(function(n){
+                        return m('.fangorn-canDelete.text-success', n.data.name);
+                    }),
+                    noDeleteList.map(function(n){
+                        return m('.fangorn-noDelete.text-warning', n.data.name);
+                    })
+                ]);
+            mithrilButtonsMultiple =  m('div', [
+                    m('span.tb-modal-btn', { 'class' : 'text-primary', onclick : function() { cancelDelete(); } }, 'Cancel'),
+                    m('span.tb-modal-btn', { 'class' : 'text-danger', onclick : function() { runDeleteMultiple.call(tb, deleteList); }  }, 'Delete Some')
+                ]);
         }
+        tb.modal.update(mithrilContentMultiple, mithrilButtonsMultiple);
     }
 
+    return true; // Let fangorn know this config option was used.
+}
+
+
+// Define Fangorn Button Actions
+function _githubDefineToolbar (item){
```

Comment:
There is a lot of logic/mixed concerns in these `defineToolbar` functions. Given that Treebeard uses mithril, why not allow add-ons to define a module for the toolbar.

```js
var GitHubToolbar = {
    controller: function(args) {
        this.onClickUpload = ...
    },
    view: function(ctrl) { 
        return m(..., onclick: ctrl.onClickUpload)
    }
}

Fangorn.config.github = {
    // ...
    toolbarModule: GitHubToolBar
}
```

This will also make it easier to break out shared components for re-use across addons.